### PR TITLE
Keylogging

### PR DESCRIPTION
**Pull request by sebag90**
keylogging plugin to log every keystroke of the users in slurk logs

### CODE REVIEW

**Comment by sebag90** [comment]
@janagoetze this pull requests adds a plug-in (keylogger) to log every single keystroke and timestamp while they chat on slurk

File: slurk/views/chat/events.py [hunk 1]
```diff
@@ -66,6 +66,14 @@ def bounding_box(payload):
             )
 
 
+@socketio.event
+def keystroke(key):
+    db = current_app.session
+    room = db.query(Room).get(key["room"]) if "room" in key else None
```
Comment (janagoetze) [inline slurk/views/chat/events.py:72]: when would room be None?

Comment (janagoetze) [reply slurk/views/chat/events.py:72]: and does it make sense to log in that case?

**Review by janagoetze — CHANGES_REQUESTED** [review]
We need to also include documentation here: https://github.com/clp-research/slurk/blob/master/docs/slurk_layouts.rst

File: slurk/views/chat/events.py (hunk 1 above)
Comment (sebag90) [reply slurk/views/chat/events.py:72]: I'm pretty sure that it shouldn't be possible, but I saw this check in the ```mouse``` event, but you're right, if we cannot match it to a room there is no reason to log it, I'm going to change it

**Review by janagoetze — APPROVED** [review]

**Pull request merged**